### PR TITLE
[4.4] Form Text Field: Translate text for frontend validation

### PR DESCRIPTION
**Pull request by Hackwar**
Pull Request for Issue #42919 .

### Summary of Changes
The validation text for frontend validation right now is not translated.

### Testing Instructions
Codereview

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by dgrammatiko** [comment]
The way to set the `data` attributes of an element should be done through the existing API. The existing code predates the API...
In sort this should be done differently

**Comment by Hackwar** [comment]
That might be, however that would be a refactoring and a new feature, which we can only really do in a major or minor release, not in a patch release. Fixing the bug for now however would be possible for 4.4, which means people on 4.4 and 5.1 would both benefit from this now instead of having to wait for 5.2.

**Comment by dgrammatiko** [comment]
I'm not sure what's the new feature here. All I said is that instead changing this in the layout you should doit in the form class. The same conditional, essentially the same code, but in the right place using the existing API for the data.
Anyways, monkey patching is also fine...

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on ebd2d9ff6170cab3dbc14e0d4e7d96b4046dcbf8<br /><br />tested by adding 			validationtext="JGLOBAL_TITLE" to the title field of administrator\components\com_content\forms\article.xml
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43198">issues.joomla.org/tracker/joomla-cms/43198</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43198">issues.joomla.org/tracker/joomla-cms/43198</a>.</sub>

File: layouts/joomla/form/field/text.php [hunk 1]
```diff
@@ -95,7 +95,7 @@
     !empty($pattern) ? 'pattern="' . $pattern . '"' : '',
 
     // @TODO add a proper string here!!!
-    !empty($validationtext) ? 'data-validation-text="' . $validationtext . '"' : '',
+    !empty($validationtext) ? 'data-validation-text="' . Text::_($validationtext) . '"' : '',
```
Comment (Fedik) [inline layouts/joomla/form/field/text.php]: The text must be escaped `$this->escape(Text::_($validationtext))`

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on ebd2d9ff6170cab3dbc14e0d4e7d96b4046dcbf8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43198">issues.joomla.org/tracker/joomla-cms/43198</a>.</sub>

**Comment by brianteeman** [comment]
changed my test result as it needs to be escaped as Fedir pointed out.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43198">issues.joomla.org/tracker/joomla-cms/43198</a>.</sub>

**Comment by Hackwar** [comment]
@Quy @brianteeman I escaped the string. Would you be able to test this again?

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 8baa9319d20e0eb37406a8bbe8fe014b4f29363c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43198">issues.joomla.org/tracker/joomla-cms/43198</a>.</sub>

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 8baa9319d20e0eb37406a8bbe8fe014b4f29363c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43198">issues.joomla.org/tracker/joomla-cms/43198</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43198">issues.joomla.org/tracker/joomla-cms/43198</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**